routers/analytics.py

```
@@ -0,0 +1,1653 @@
+                percentage = 0.0 if len(current_records) == 0 else 100.0
+            else:
+                percentage = round(((len(current_records) - len(previous_records)) / len(previous_records)) * 100, 1)
+            
+            return JSONResponse(content={"detail": results, "percentage": percentage}, status_code = 200)
+        else:
+            raise HTTPException(status_code = 404, detail = "An error occurred: bot doesn\'t exist")
+
+    except HTTPException as e:
+            raise e
+    except Exception as e:
+        raise HTTPException(status_code = 500, detail = f"An error occurred: {str(e)}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)